File: 준영/소수 찾기.py
def solution(n):

    sieve = [True]*(n+1)

    m = int(n ** 0.5)
    for i in range(2, m+1):
        if sieve[i] == True:
            for j in range(i*i, n+1, i):
                sieve[j] = False

    x = [i for i in range(2, n+1) if sieve[i] == True]
    
    answer = len(x)
    return answer

File: 준영/소수 찾기.js
function solution(n) {
  var answer = 0
  var arr = Array.from({length: n+1}, x => true)
  var m = parseInt(n ** 0.5)
  for (let i = 2; i < m+1; i++) {
    if (arr[i] === true) {
      for (let j = i*i; j < n+1; j += i) {
        arr[j] = false
      }
    }
  }
  answer = arr.reduce((acc, cur) => {
    if (cur === false) {
      return acc
    } else {
      return acc + 1
    }
  }, -2)

  return answer 
}

solution(10)